src/api/flow.ts

```diff
@@ -10,32 +10,9 @@ export interface FlowIterator<T> {
     throw?(e?: any): IteratorResult<T> | Promise<IteratorResult<T>>
 }
 
-export function flow<R>(generator: () => FlowIterator<any>): () => CancellablePromise<R>
-export function flow<A1>(
-    generator: (a1: A1) => FlowIterator<any>
-): (a1: A1) => CancellablePromise<any> // Ideally we want to have R instead of Any, but cannot specify R without specifying A1 etc... 'any' as result is better then not specifying request args
-export function flow<A1, A2>(
-    generator: (a1: A1, a2: A2) => FlowIterator<any>
-): (a1: A1, a2: A2) => CancellablePromise<any>
-export function flow<A1, A2, A3>(
-    generator: (a1: A1, a2: A2, a3: A3) => FlowIterator<any>
-): (a1: A1, a2: A2, a3: A3) => CancellablePromise<any>
-export function flow<A1, A2, A3, A4>(
-    generator: (a1: A1, a2: A2, a3: A3, a4: A4) => FlowIterator<any>
-): (a1: A1, a2: A2, a3: A3, a4: A4) => CancellablePromise<any>
-export function flow<A1, A2, A3, A4, A5>(
-    generator: (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5) => FlowIterator<any>
-): (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5) => CancellablePromise<any>
-export function flow<A1, A2, A3, A4, A5, A6>(
-    generator: (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6) => FlowIterator<any>
-): (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6) => CancellablePromise<any>
-export function flow<A1, A2, A3, A4, A5, A6, A7>(
-    generator: (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6, a7: A7) => FlowIterator<any>
-): (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6, a7: A7) => CancellablePromise<any>
-export function flow<A1, A2, A3, A4, A5, A6, A7, A8>(
-    generator: (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6, a7: A7, a8: A8) => FlowIterator<any>
-): (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5, a6: A6, a7: A7, a8: A8) => CancellablePromise<any>
-export function flow(generator: Function) {
+export function flow<T, U extends any[]>(
+    generator: (...args: U) => FlowIterator<any>
+): CancellablePromise<U> {
```

Comment:
Why `<U>`?

Reply:
No significant meaning here. It can also be changed to <R, A> to follow the previous naming.

Reply:
No, it cannot. Right result type is `(...args: U) => CancellablePromise<any>`

Reply:
You're right, I missed the `() =>` part. Strange that no tests are failing.